### ISSUE TITLE
If it's isomorphic code, why do you require different code on the server

### ISSUE THREAD

**Issue by zwhitchcox** [facebook/react#5583]
In the [docs](https://www.npmjs.com/package/react-dom), you have an example for server side code and an example for client side code. I thought isomorphic code was supposed to work on both automatically.

**Comment by jimfb** [comment]
You don't have a DOM on the server side (you only have the html markup that is being sent to the client).  Thus, you need to render-to-string instead of render-to-dom.

Isomorphic code generally allows for lots of code reuse, but there are always going to be edge cases.  For instance, your datastore implementations might be slightly different to account for caching/security differences on server/client.  You'll also need to do a little wiring to revive your markup on the client side (ie. the `ReactDOM.render`), as you pointed out.

This is a usage question, rather than a bug in the React core. Usage questions are better answered on sites like StackOverflow, as we try to use github issues for tracking bugs in the React core. For this reason, I'm going to close out this issue, but feel free to continue the conversation here or move it to StackOverflow.

**Comment by zwhitchcox** [comment]
Ok, I'm kind of confused now. Is there any detailed docs for this npm package? I just noticed, this took me to the main react library not the npm package. I'm a little confused now.

**Comment by jimfb** [comment]
I'm not sure what you mean by "detailed docs for this npm package".  The docs are the same for React with/without npm.

If you are just getting started with React, and haven't used npm before, the npm stuff often adds unnecessary confusion/complication.  You might try downloading the starter pack and playing with the code there: http://facebook.github.io/react/downloads/react-0.14.3.zip

Once you get the hang of React and how to write a simple React app, then you can worry about making your app isomorphic and setting up your infrastructure.

**Comment by zwhitchcox** [comment]
No, I mean, I'm comfortable with npm and React, I guess it's just confusing to me that there's a main React package and then there's another React part inside another package. It seems like `react-dom` would just be included in React.

So, what I'm saying is if this is a part of React, then why is it in a separate npm package. There's a main `react` package, so why doesn't that include the `react-dom`? It's just confusing to me, although, I think I kind of understand now.

**Comment by iamdustan** [comment]
@zwhitchcox this may help: https://facebook.github.io/react/blog/2015/07/03/react-v0.14-beta-1.html#two-packages

tldr; There is React core and then there are renderers such as react-native and react-dom (official) and react-blessed, react-art, react-canvas, etc.
